repo-agent/pkg/agentoutput/watcher.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2024 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package agentoutput
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	outputFile = "/workspaces/agent-output.txt"
+)
+
+// Run starts the sidecar watcher loop.
+// componentName is used for logging (e.g., "issue", "review").
+// gvr is the GroupVersionResource to update.
+func Run(componentName string, gvr schema.GroupVersionResource) {
+	fmt.Printf("starting %s sidecar\n", componentName)
+	name := os.Getenv("NAME")
+	if name == "" {
+		fmt.Println("missing NAME env")
+		os.Exit(1)
+	}
+	namespace := os.Getenv("NAMESPACE")
+	if namespace == "" {
+		fmt.Println("missing NAMESPACE env")
+		os.Exit(1)
```

Comment:
Calling `panic` is not idiomatic for error handling in Go applications. It should be reserved for truly unrecoverable situations. It's better to return the error and let the calling function handle it.